File: 10105 Polynomial Coefficients/solution.java
import java.io.*;

public class Main {
    static int[] fac = {1, 1, 2, 6, 24, 120, 720, 5040, 40320, 362880, 3628800, 39916800, 479001600};
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StreamTokenizer st = new StreamTokenizer(br);
        StringBuilder output = new StringBuilder();
        while(st.nextToken() == StreamTokenizer.TT_NUMBER) {
            int ans = fac[(int)st.nval];
            st.nextToken();
            int k = (int)st.nval;
            while(k-- > 0) {
                st.nextToken();
                ans /= fac[(int)st.nval];
            }
            output.append(ans).append('\n');
        }
        System.out.print(output);
    }
}

File: 10105 Polynomial Coefficients/solution.c
#include <stdio.h>

int fac[] = {1, 1, 2, 6, 24, 120, 720, 5040, 40320, 362880, 3628800, 39916800, 479001600};

int main() {
    int n, k;
    while(scanf("%d %d", &n, &k) == 2) {
        int ans = fac[n];
        while(k--) {
            int nk;
            scanf("%d", &nk);
            ans /= fac[nk];
        }
        printf("%d\n", ans);
    }
    return 0;
}
